core/types/system_tx.go

```diff
@@ -0,0 +1,87 @@
+package types
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+type SystemTx struct {
+	From  common.Address  // pre-determined sender
```

Comment:
You need to modify this part to be able to use `From` as the sender: https://github.com/scroll-tech/go-ethereum/blob/develop/core/types/transaction_signing.go#L257

Please search for all occurrences of `IsL1MessageTx` and `AsL1MessageTx` and consider if we need to add system txs.

Reply:
ah, thanks.

Reply:
fixed.

Reply:
do we need to do anything here: https://github.com/scroll-tech/go-ethereum/blob/733dd01a9af05c78f617731127a1aeae93a6cd0b/rollup/types/encoding/da.go#L225 ?

Reply:
Let's rename the variable to `Sender` just to be consistent with `L1MessageTx`.

Reply:
> do we need to do anything here

Yes, we need to add proper encoding of `SystemTx` to `TransactionData` (in `l2trace.go`) and decide how it's encoded in our DA commit transactions. Please note these down, but we can add them separately.